flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartition.java

```diff
@@ -77,6 +78,10 @@
 	/** Flag indicating whether the subpartition has been released. */
 	private volatile boolean isReleased;
 
+	/** The number of non-event buffers currently in this subpartition */
+	@GuardedBy("buffers")
+	private volatile int buffersInBacklog;
```

Comment:
If the interface of `getNextBuffer()` was changed as suggested above, we could remove the `volatile` here as well.